core/core.go

```diff
@@ -343,9 +344,15 @@ func NewSinkSession[I BlockId, F Flags](
 		store,
 		util.NewSynchronizedDeque[Block[I]](util.NewBlocksDeque[Block[I]](2048)),
 		stats,
+		make(chan error, 1),
```

Comment:
I had to re-read about channel capacity in golang... with a buffer size of 1 - a second error sent here will block, which is fine because any error coming across this channel means something failed and we're done. More or less?

Reply:
I'm quite new to this, but my thinking was...
1) Sends to unbuffered channels block, and I want to ensure that if we error before we start selecting over the done channel, the sends don't block.
2) It is good to be explicit about how many of something we need vs having things be unbounded, and as coded there should only ever be a single error, since we always return after sending an error.
3) While I see a mix of buffered and unbuffered in the Go code base, it seems that most of what I see there is buffered of size 1.